### DISCUSSION TITLE
Query + relations

### DISCUSSION THREAD

**Discussion by DrWaan** [ramnes/notion-sdk-py#114] [Q&A] [answered]
Hi,
can you help me with the code for query a database and filtering by relations? I tried it with the code bellow. Filtering by multi selects works fine only quering by relations doesn't work.
```python
data = client.databases.query(
         **{
             "database_id": DATABASE_ID_MY_UNI_TABLE,
             "filter": {
                 "and": [
                     {
                         "property": "NAME", "multi_select": {"contains": "XXX"}
                     },
                     {
                        "type": "relation_Name", "relation": [{"id":"ID_of_my_relation"}]
                    }
                ]
            }
        }
    )
    return MyData(data=data)
```

i get an Error type like this:  raise APIResponseError(response, body["message"], code)
notion_client.errors.APIResponseError: body failed validation. Fix one: body.filter.and[1].title should be defined, ....

while
```python
data = client.databases.query(
         **{
             "database_id": DATABASE_ID_MY_UNI_TABLE,
             "filter": {
                 "and": [
                     {
                         "property": "NAME", "multi_select": {"contains": "XXX"}
                     }
                ]
            }
        }
    )
    return MyData(data=data)
```
works fine.

It can't be an ID error or something else

**Comment by DrWaan** [comment]
i also trid
`                     {
                        "property": "relation_Name", "relation": [{"id":"ID_of_my_relation"}]
                    }`
and
`
                     {
                       "property": "relation_Name", "relation": {"is_empty":"true"}
                    }
`

both results in an error.
Can anyone help?

**Reply by kbruxvoort** [reply, marked as answer]
In order to use the ID in the query you need to pass "contains" or "does_not_contain"
```
"filter": {
    "property": relation_name,
    "relation": {
        "contains": relation_id
    }
}
```

The second filter is failing because you are passing true as a string instead of a boolean
```
"filter": {
    "property": relation_name,
    "relation": {
        "is_empty": True
    }
}
```

**Reply by DrWaan** [reply]
Thanks a lot!

**Comment by jg33** [comment]
I'm getting a lot of errors like this: 
```
results -> 0 -> properties -> On Worker -> type
  unexpected value; permitted: <PropertyValueType.FORMULA: 'formula'> (type=value_error.const; given=relation; permitted=[<PropertyValueType.FORMULA: 'formula'>])
```
when I query a db that has a relationship in it. It looks like it's expecting a PropertyValueType it's not expecting. any idea what might be happening? 

Thanks!
